lib/b-tree.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+
+function BTree(degree = DEFAULT_DEGREE) {
+  this.root = new BTreeNode();
+  this.minDegree = degree;
+}
+
+function BTreeNode(leaf = true) {
+  this.elements = [];
+  this.leaf = leaf;
+  this.lastChild = null;
+}
+
+function Element(value, typle) {
+  this.value = value;
+  this.typle = typle;
+  this.child = null;
+}
+
+BTree.prototype.getEqual = function(value) {
+  let curNode = this.root;
+  while (true) {
+    let needChangedNodeFlag = true;
+    for (const curElement of curNode.elements) {
+      if (value <= curElement.value) {
+        if (curElement.value === value) return curElement.typle;
+        if (curNode.leaf) return null;
```

Comment:
Why check for `leaf` twice (here and blow on line 36)?

Reply:
These are different blocks. The first block for nodes with values not less than the specified, and on line 36 for those with a value less than